nabclockd/nabclockd.py

```diff
@@ -141,10 +162,53 @@ async def clock_loop(self):
                         response = self.clock_response(now)
                         for r in response:
                             if r == "sleep":
+                                # Check if we need to play the sleep sound
+                                if self.config.play_wakeup_sleep_sounds:
+                                    idle_elapsed_seconds = (
+                                        datetime.datetime.now()
+                                        - self.last_time_idle_state
+                                    ).total_seconds()
+
+                                    # Skip the sound if less than 1 second
+                                    # has elapsed since the last idle packet,
+                                    # that means it went to sleep
+                                    # right after boot
+                                    if idle_elapsed_seconds > 1:
+                                        # We dont want the next idle packet
+                                        # that is sent after the sound to
+                                        # trigger a loop-cv-notify, so skip it
+                                        self.ignore_next_idle_packet = True
+                                        packet = (
+                                            '{"type":"message",'
+                                            '"body":[{"audio":["sleep/*.mp3"],'
+                                            '"choreography":null}],'
+                                            '"request_id":"sleep_sound"}\r\n'
+                                        )
+                                        self.writer.write(
+                                            packet.encode("utf8")
+                                        )
+                                        await self.writer.drain()
+
                                 self.writer.write(b'{"type":"sleep"}\r\n')
                                 await self.writer.drain()
                                 self.asleep = None
+
                             elif r == "wakeup":
+                                # Check if we need to play the wakeup sound
+                                if self.config.play_wakeup_sleep_sounds:
+                                    # We dont want the next idle packet
+                                    # that is sent after the sound to
+                                    # trigger a loop-cv-notify, so skip it
+                                    self.ignore_next_idle_packet = True
```

Comment:
After a sound is played, an additional status packet is received which was disturbing the loop processing when a wakeup/sleep event is received.
I added this flag to force the next idle packet to be skipped by  'process_nabd_packet' so that it would not trigger a loop-notify event only for the next packet received.